setup/setup.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const root = process.cwd();
+const wpfolder = path.resolve(root, "node_modules/webpack/");
+const msg = {
+	setupStart: "\nSetup: start\n\r",
+	setupComplete: "Setup: complete\n\r",
+	setupNoSymlink: "\nSetup: no symlink\n\r",
+	setupSymlink: "\nSetup: symlink established\n\r",
+	setupFail: "\nSetup: failed\n\r",
+	setupSkip: "\nSetup: skip\n\r",
+	setupInstallYarn: "\nSetup: Installing Yarn\n\r",
+	setupInstallDependencies: "\nSetup: Installing dependencies\n\r"
+};
+
+function setup() {
+	return new Promise((resolve, reject) => {
+		checkSymlinkExistsAsync()
+			.then(() => {
+				resolve(msg.setupSkip);
+			})
+			.catch(() => {
+				resolve(
+					ensureYarnInstalledAsync().then(() => {
+						return runSetupAsync().then(() => {
+							return checkSymlinkExistsAsync();
+						});
+					})
+				);
+			});
+	})
+		.then(message => {
+			console.log(message);
+			message !== msg.setupSkip && console.log(msg.setupComplete);
+			process.exitCode = 0;
+		})
+		.catch(e => {
+			console.log(e);
+			process.exitCode = 1;
+		});
+}
+
+function runSetupAsync() {
+	console.log(msg.setupInstallDependencies);
+	return new Promise((resolve, reject) => {
+		let cp = require("child_process").exec(
```

Comment:
Move `require` to the top of the file:
`const { exec } = require('child_process')`

Reply:
I had put them inside the func, because they are not always needed, for performance reasons.
As the script will also be called when executing `npm run test` or `yarn test`, but perhaps it's not noticable, if we load `fs` anyway...


Done.